### ISSUE TITLE
[Epic] Migrate from formik to react-hook-form

### ISSUE THREAD

**Issue by flash1293** [airbytehq/airbyte#24291]
Refactor our form handling and migrate to a better supported library to ensure consistency, performance and maintainability

Tech spec: https://docs.google.com/document/d/1kU8CpqNKZQY91gTKc2BtcKRA021Kc9mrwgPrbvXhgoQ/edit#heading=h.fa72b39y2m99

## Phase 1
* Prepare first iteration of library form components
* Migrate connector extensibility forms
* Iterate on library form components and document gotchas / common cases

## Phase 2
* In parallel: All others teams migrates the forms owned by them
* Use standard form components where possible

**Comment by timroes** [comment]
Duplicate of https://github.com/airbytehq/airbyte/issues/24246
